src/test/kotlin/uk/gov/dluhc/printapi/database/repository/AnonymousElectorDocumentSummaryRepositoryIntegrationTest.kt

```diff
@@ -0,0 +1,222 @@
+
+            val aed1InitialDocument = buildAnonymousElectorDocument(
+                gssCode = gssCode,
+                sourceReference = aed1SourceReference,
+                applicationReference = aed1ApplicationReference,
+                issueDate = currentDate.minusDays(10),
+                requestDateTime = currentDateTimeInstant.minusSeconds(10),
+                contactDetails = buildAedContactDetails(surname = "A")
+            )
+            val aed1SecondDocument = buildAnonymousElectorDocument(
+                gssCode = gssCode,
+                sourceReference = aed1SourceReference,
+                applicationReference = aed1ApplicationReference,
+                issueDate = currentDate.minusDays(10),
+                requestDateTime = currentDateTimeInstant.minusSeconds(9),
+                contactDetails = aed1InitialDocument.contactDetails!!
+            )
+            val aed1LatestDocument = buildAnonymousElectorDocument(
+                gssCode = gssCode,
+                sourceReference = aed1SourceReference,
+                applicationReference = aed1ApplicationReference,
+                issueDate = currentDate.minusDays(9),
+                requestDateTime = currentDateTimeInstant, // View will return this latest record
+                contactDetails = aed1InitialDocument.contactDetails!!
+            )
+
+            val aed2SourceReference = aValidSourceReference()
+            val aed2ApplicationReference = aValidApplicationReference()
+            val aed2InitialDocument = buildAnonymousElectorDocument(
+                gssCode = gssCode,
+                sourceReference = aed2SourceReference,
+                applicationReference = aed2ApplicationReference,
+                issueDate = currentDate.minusDays(10),
+                requestDateTime = currentDateTimeInstant.minusSeconds(10),
+                contactDetails = buildAedContactDetails(surname = "Z")
+            )
+            val aed2SecondDocument = buildAnonymousElectorDocument(
+                gssCode = gssCode,
+                sourceReference = aed2SourceReference,
+                applicationReference = aed2ApplicationReference,
+                issueDate = currentDate.minusDays(9),
+                requestDateTime = currentDateTimeInstant.minusSeconds(8),
+                contactDetails = aed2InitialDocument.contactDetails!!
+            )
+            val aed2LatestDocument = buildAnonymousElectorDocument(
+                gssCode = gssCode,
+                sourceReference = aed2SourceReference,
+                applicationReference = aed2ApplicationReference,
+                issueDate = currentDate.minusDays(9),
+                requestDateTime = currentDateTimeInstant, // View will return this latest record
+                contactDetails = aed2InitialDocument.contactDetails!!
+            )
+            val aed3Document = buildAnonymousElectorDocument(gssCode = gssCode, issueDate = currentDate.plusDays(10))
```

Comment:
plusDays(10) seems odd since it will never happen in practice?

Reply:
I have renamed the variables now. This is for another application that was added after 10 days, so is very much possible. I think the confusion could have been because the variable name was earlier renamed with "world class" standards 😆